lib/git-shell-out-strategy.js

```diff
@@ -415,6 +415,39 @@ export default class GitShellOutStrategy {
     return this.exec(args, {writeOperation: true});
   }
 
+  async getCommitMessageTemplate() {
+    let templatePath = await this.getConfig('commit.template');
+    if (!templatePath) {
+      return null;
+    }
+
+    /**
+     * Get absolute path from git path
+     * https://git-scm.com/docs/git-config#git-config-pathname
+     */
+    const homeDir = os.homedir();
+    // this regex attempts to get the specified user's home directory
+    // Ex: on Mac ~kuychaco/ is expanded to the specified user’s home directory (/Users/kuychaco)
+
+    // Regex translation:
+    // ^~ line starts with tilde
+    // ([^/]*)/ captures non-forwardslash characters before first slash
+    const regex = new RegExp('^~([^/]*)/');
+    templatePath = templatePath.trim().replace(regex, (_, user) => {
+      // if no user is specified, fall back to using the home directory.
+      return `${user ? path.join(path.dirname(homeDir), user) : homeDir}/`;
```

Comment:
@smashwilson: can you test this on non-Mac operating systems?  We want to make sure if a username is specified, it does the right thing for our Linux and or Windows users.